Vehicle_Parking_Management.tests/Test/Controller/VehicleControllerTest.cs

```diff
@@ -0,0 +1,123 @@
+using AutoFixture;
+using FluentAssertions;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using Vehicle_Parking_Management.Models.Api;
+using VehicleManagementSystem.Models.Data;
+using VehicleManagementSystem.Services.Vehicle;
+
+namespace Vehicle_Parking_Management.tests.Test.Controller
+{
+    public class VehicleControllerTest
+    {
+        private readonly Fixture _fixture = new Fixture();
+        [Fact]
+        public async Task ShouldReturnOn_GetAllVehicle_AndStatusAs200()
+        {
+            var mockVehicleService = new Mock<IVehicleService>();
+            var mockVehicle=_fixture.Create<List<VehicleModel>>();
+            mockVehicleService.Setup(service => service.getAllVehicles()).
+                ReturnsAsync(mockVehicle);
+
+            var testController = new VehicleController(mockVehicleService.Object);
+            var result = await testController.GetAllVehicle();
+
+            mockVehicleService.Verify(service => service.getAllVehicles(), Times.Once());
+            result.Should().BeOfType<OkObjectResult>();
+            var objectResult = (OkObjectResult)result;
```

Comment:
use and - please go through docs of fluent assertions 
line 26 returns OkObjectResult

Reply:
resolved.